src/tfp_backend/Code_gen.ml

```diff
@@ -103,17 +103,35 @@ let pp_init ppf p =
   let pp_save_data ppf (name, st) =
     pf ppf "self.%s = %a" name (pp_cast "") (name, st)
   in
+  let pp_prep_data_stmt ppf st = pf ppf "self.%a" pp_stmt st in
   let ppbody ppf =
     match p.Program.input_vars with
     | [] -> pf ppf "pass"
-    | _ -> (list ~sep:cut pp_save_data) ppf p.Program.input_vars
+    | _ ->
+        pf ppf "@[<v>%a@,%a@]"
+          (list ~sep:cut pp_save_data)
+          p.Program.input_vars
+          (list ~sep:cut pp_prep_data_stmt)
+          p.Program.prepare_data
   in
   pp_method ppf "__init__" ("self" :: List.map ~f:fst p.input_vars) [] ppbody
 
 let pp_extract_data ppf p =
   let pp_data ppf (name, _) = pf ppf "%s = self.%s" name name in
   (list ~sep:cut pp_data) ppf p.Program.input_vars
 
+let pp_transformed_data ppf p =
```

Comment:
This is just printing out the transformed data var reassignments, right? Could we change the name to reflect that (and possibly share code with the above equivalent for data variables?

Reply:
Renamed the functions to be more consistent with naming throughout the file, and added `pp_var_assignment` function that can is called from `pp_extract_data` and `pp_transformed_data`.